import/plugin.cpp

```diff
@@ -84,52 +43,31 @@ void SailfishOSWebViewPlugin::initializeEngine(QQmlEngine *engine, const char *u
-        // correctly.
-        mozCtxt->setPref("layers.low-precision-buffer", QVariant::fromValue<bool>(false));
-        // Standard settings.
-        mozCtxt->setPref("gfx.qt.rgb16.force", QVariant::fromValue<bool>(false));
-        mozCtxt->setPref("embedlite.azpc.handle.singletap", QVariant::fromValue<bool>(false));
-        mozCtxt->setPref("embedlite.azpc.json.singletap", QVariant::fromValue<bool>(true));
-        mozCtxt->setPref("embedlite.azpc.handle.longtap", QVariant::fromValue<bool>(false));
-        mozCtxt->setPref("embedlite.azpc.json.longtap", QVariant::fromValue<bool>(true));
-        mozCtxt->setPref("embedlite.azpc.json.viewport", QVariant::fromValue<bool>(true));
-        mozCtxt->setPref("apz.asyncscroll.throttle", QVariant::fromValue<int>(40));
-        mozCtxt->setPref("apz.asyncscroll.timeout", QVariant::fromValue<int>(400));
-        mozCtxt->setPref("apz.fling_stopped_threshold", QLatin1String("0.13f"));
-
-        // Don't expose any protocol handlers by default and don't warn about those.
-        mozCtxt->setPref(QStringLiteral("network.protocol-handler.external-default"), false);
-        mozCtxt->setPref(QStringLiteral("network.protocol-handler.expose-all"), false);
-        mozCtxt->setPref(QStringLiteral("network.protocol-handler.warn-external-default"), false);
-
-        int screenWidth = QGuiApplication::primaryScreen()->size().width();
-        int tileSize = screenWidth;
-
-        // With bigger than qHD screen fill with two tiles in row (portrait).
-        // Landscape will be filled with same tile size.
-        if (screenWidth > 540) {
-            tileSize = screenWidth / 2;
-        }
-
-        mozCtxt->setPref(QStringLiteral("layers.tile-width"), QVariant(tileSize));
-        mozCtxt->setPref(QStringLiteral("layers.tile-height"), QVariant(tileSize));
-    }
+    SailfishOS::WebEngine *webEngine = SailfishOS::WebEngine::instance();
+
+    SailfishOS::WebEngineSettings::initialize();
+    SailfishOS::WebEngineSettings *engineSettings = SailfishOS::WebEngineSettings::instance();
+
+
+    // For some yet unknown reason QmlMozView crashes when
+    // flicking quickly if progressive-paint is enabled.
+    engineSettings->setPreference("layers.progressive-paint", QVariant::fromValue<bool>(false));
+    // Disable low-precision-buffer so that background underdraw works
+    // correctly.
+    engineSettings->setPreference("layers.low-precision-buffer", QVariant::fromValue<bool>(false));
+
+    // Don't expose any protocol handlers by default and don't warn about those.
+    engineSettings->setPreference(QStringLiteral("network.protocol-handler.external-default"), false);
+    engineSettings->setPreference(QStringLiteral("network.protocol-handler.expose-all"), false);
+    engineSettings->setPreference(QStringLiteral("network.protocol-handler.warn-external-default"), false);
+
+    engine->rootContext()->setContextProperty("MozContext", webEngine);
```

Comment:
We can already create instances of WebEngine/WebEngineSettings that will share private implementation. Problem that I'm currently seeing is commented in https://github.com/sailfishos/sailfish-components-webview/pull/7#pullrequestreview-697748 e.i. how to deal initialization?

Maybe need to just rid off initialization arguments and handle custom cases separately.